public/app/core/components/Select/DashboardPicker.tsx

```diff
@@ -8,64 +9,40 @@ import { DashboardSearchHit, DashboardDTO } from 'app/types';
 
 export interface Props {
   onSelected: (dashboard: DashboardDTO) => void;
-  currentDashboardId?: SelectableValue<number>;
+  currentDashboard?: SelectableValue<number>;
   size?: FormInputSize;
+  isClearable?: boolean;
 }
 
-export interface State {
-  isLoading: boolean;
-}
-
-export class DashboardPicker extends PureComponent<Props, State> {
-  debouncedSearch: any;
-
-  static defaultProps = {
-    size: 'md',
-  };
-
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
-      isLoading: false,
-    };
-
-    this.debouncedSearch = debounce(this.getDashboards, 300, {
-      leading: true,
-      trailing: true,
-    });
-  }
-
-  getDashboards = (query = '') => {
-    this.setState({ isLoading: true });
+export const DashboardPicker: FC<Props> = ({ onSelected, currentDashboard, size = 'md', isClearable = false }) => {
+  const getDashboards = (query = '') => {
```

Comment:
Think this can be defined outside of the component not to create a new function with every render